docs/source/kto_trainer.mdx

```diff
@@ -0,0 +1,74 @@
+# KTO Trainer
+
+TRL supports the Kahneman-Tversky optimization (KTO) Trainer for training language models from unpaired preference data, as described in the [report](https://github.com/ContextualAI/HALOs/blob/main/assets/report.pdf) by Kawin Ethayarajh, Winnie Xu, Dan Jurafsky, and Douwe Kiela.
+
+## Expected dataset format
+
+The KTO trainer expects a very specific format for the dataset as it does not require pairwise preferences. Since the model will be trained to directly optimize un-paired datasets of preferred and un-preferred generations, given some prompts, we expect a dataset with the following columns:
+
+- `prompt`
+- `generation`
+- `chosen`
```

Comment:
would `label` be a better choice here since in principle we may have a scalar score?
```suggestion
- `label`
```